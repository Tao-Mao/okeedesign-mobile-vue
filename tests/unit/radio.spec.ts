import { mount, later } from '../index';
import { Radio as RadioNative, RadioGroup as RadioGroupNative } from '@src';
const Radio = RadioNative as any;
const RadioGroup = RadioGroupNative as any;

function renderWrapper(options = {} as any) {
  return mount({
    template: `
      <o-radio-group v-model="radio" :disabled="disabled" :stopPropagation="stopPropagation" :size="size">
        <o-radio value="1">单选框 1</o-radio>
        <o-radio value="2">单选框 2</o-radio>
        <o-radio value="3">单选框 3</o-radio>
      </o-radio-group>
    `,
    data() {
      return {
        radio: '1',
        disabled: options.disabled || false,
        size: options.size || 'normal',
        stopPropagation: options.stopPropagation || false,
      };
    },
  });
}

describe('radio', () => {
  test('render basic Usage', () => {
    const wrapper = renderWrapper();
    expect(wrapper).toMatchSnapshot();
  });
  test('render disabled Usage', () => {
    const wrapper = renderWrapper({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });
  test('not trigger click event when disabled', async () => {
    const onClick = jest.fn();
    const onChange = jest.fn();
    const wrapper = mount({
      template: `
        <o-radio-group v-model="radio" @click="onClick" @change="onChange">
          <o-radio value="1" :disabled="disabled">单选框 1</o-radio>
          <o-radio value="2">单选框 2</o-radio>
          <o-radio value="3">单选框 3</o-radio>
        </o-radio-group>
      `,
      components: {
        'o-radio-group': RadioGroup,
        'o-radio': Radio,
      },
      data() {
        return {
          radio: '2',
          disabled: true,
        };
      },
      methods: {
        onClick,
        onChange,
      },
    }) as any;
    await later();
    const radios = wrapper.findAll('.o-radio');
    await radios.at(0).trigger('click');
    expect(onClick).toBeCalledTimes(0);
    await radios.at(2).trigger('click');
    expect(onClick).toBeCalledTimes(1);
    expect(onChange).toBeCalledTimes(1);
  });
});
