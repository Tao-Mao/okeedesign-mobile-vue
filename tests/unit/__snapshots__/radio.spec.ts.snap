// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio render basic Usage 1`] = `<div class="o-radio-group"><label class="o-radio o-radio--normal  o-radio--checked"><i class="o-icon o-icon--check-one o-radio__icon">test-file-stub</i><span class="o-radio__label">单选框 1</span><input type="radio" class="o-radio__input" value="1"></label> <label class="o-radio o-radio--normal"><i class="o-radio__icon"></i><span class="o-radio__label">单选框 2</span><input type="radio" class="o-radio__input" value="2"></label> <label class="o-radio o-radio--normal"><i class="o-radio__icon"></i><span class="o-radio__label">单选框 3</span><input type="radio" class="o-radio__input" value="3"></label></div>`;

exports[`radio render disabled Usage 1`] = `<div class="o-radio-group"><label class="o-radio o-radio--normal  o-radio--disabled o-radio--checked"><i class="o-icon o-icon--check-one o-radio__icon">test-file-stub</i><span class="o-radio__label">单选框 1</span><input type="radio" disabled="disabled" class="o-radio__input" value="1"></label> <label class="o-radio o-radio--normal  o-radio--disabled"><i class="o-radio__icon"></i><span class="o-radio__label">单选框 2</span><input type="radio" disabled="disabled" class="o-radio__input" value="2"></label> <label class="o-radio o-radio--normal  o-radio--disabled"><i class="o-radio__icon"></i><span class="o-radio__label">单选框 3</span><input type="radio" disabled="disabled" class="o-radio__input" value="3"></label></div>`;
